lhotse/cut/set.py

```diff
@@ -3562,9 +3562,15 @@ def noise_gen():
             # Keep sampling until we mixed in a "duration" amount of noise.
             # Note: we subtract 0.05s (50ms) from the target duration to avoid edge cases
             #       where we mix in some noise cut that effectively has 0 frames of features.
-            while mixed_in_duration < target_mixed_duration:
+            while mixed_in_duration < (
+                target_mixed_duration - 0.05
```

Comment:
why is the tolerance applied conditionally? shouldn’t it be unconditional here?

Reply:
Thanks. You're right. I'm not sure if we need to subtract `50ms` though (i.e., `10ms` works, too).

Reply:
Might have been an abundance of caution for feature extractors with different frame sizes and shifts. If you find it’s negatively affecting some experiments we can modify it otherwise I’d just keep it.

Reply:
makes sense. It's updated accordingly.